### ISSUE TITLE
CR:  Investigate changing specialist categorization:  "only works out of hospitals or clinics" (with blue arrow)

### ISSUE THREAD

**Issue by KelseyDH** [pathwaysmedical/frasernw#256]
At the November 2015 User group we have been tasked with investigating how to improve the categorization `only works out of hospitals or clinics`.

***Problem:  When a Specialist is given the categorization of `only works out of hospitals or clinics`, the system:***

1.  Automatically hides the following information sections from a Specialist's profile:  

  - Office Information (includes language spoken)

  - Patient Information

  - Referrals

  - Urgent Referrals

2.  Automatically removes these edit box sections from a Specialist's edit form:

  - Status (includes Language)

  - Location (office info + address + schedule)

  - Referral Details:   referral contact info, wait time

3.  Prevents a user from being able to "control" a Specialist's profile.

4.  Prevents the uploading of a Referral Form for that Specialist profile.

============
**Tracy's email:**

As discussed at last night's User Group meeting, I agreed to start an email conversation regarding how we might be able to enhance the amount of information captured for a physician who works out of a clinic, but whom we want to show more information on.  

I spent a fair bit of time looking at this today, and have some thoughts on this............

Currently, when we list a physician with a Categorization of "only works out of hospitals or clinics", the system automatically stubs down the entry.  And, although we can still enter some information for them, we are missing out on entering:

- locations 1, 2, 3, (which further opens up the options to add phone number, fax number, direct line for physician use, email addresses, website, address, open Saturday or Sunday?, Year that practice opened at this location, phone line hours, fee structure).

- We also miss out on the ability to enter MOA contact information, status details, practice limitations, languages spoken, and all the referral details, as well as patient instructions, and cancellation policy.

I wish that "only works out of clinics or hospitals" was a "status" on Pathways rather than a "categorization", as the physician may have responded to a survey, but also works out of a clinic.

![categorization_options](https://cloud.githubusercontent.com/assets/6710303/11579564/210e2dd6-99ec-11e5-965d-8e9d63888483.png)

Right now, we have to choose one or the other option, even though both may be true.  I think this would be too complicated to try to change this now, however.

My proposal is that we leave their categorization as "only works out of clinics or hospitals", but have the system NOT stub down the entry (leaving open all the same options as for "responded to survey". 

I would like to see a new option added under "Office contact information" where it shows locations one, two and three, where we could link them to "In an existing Clinic", which would further open a drop down menu to choose from clinics that are already on Pathways (as there currently is for offices).   ***The clinic would have to exist first.  You could not enter a new standalone clinic from here.

​![categorization2](https://cloud.githubusercontent.com/assets/6710303/11579570/2fc482d0-99ec-11e5-9d06-54ef88022d73.png)

Currently there is an option too choose "in a new office in a clinic".  I think this should be considered for deletion as it is confusing.  I can't think of an instance when there would be a new office in a clinic - can anyone else?  

As with selecting "in an existing office", you would still be able to enter phone numbers, etc.

I would then like to see a new "status" added,..... something to the affect of "Referrals will be triaged through the clinic" (or some wording like that) ........  This status would also have to have the blue arrow symbol associated with it, so as not to change how specialists already listed under the blue arrow are shown.

​![categorization3-status options](https://cloud.githubusercontent.com/assets/6710303/11579579/4c01171a-99ec-11e5-9e6c-5dedb7cca030.png)
​
As with other specialists on the site, I would like to be able to upload a referral form to their profile page (even though they have a blue arrow status).  Currently, it looks like you can do this, but it does not show up on their profile - I tried this recently.  This would just be to save the user from having to always go to the clinic page to access the form (as recently requested by Dr. REDACTED office/REDACTED Clinic)

And I would like to see a change to the system, allowing us to link a user account to these "blue arrow" specialists for editing, so that they can edit their own profile, which right now they cannot.  Their name does not come up to link to, when creating a new user account, because of their categorization.  User accounts for clinics are generally given to a clinic manager who has the ability to edit the clinic and specialists attached to it.  I see this as a problem because physicians may leave a clinic, but have no way of updating their profile on Pathways themselves.  Some of these physicians have access to Pathways, but cannot be linked to their own profile for editing purposes (see Dr. REDACTED) https://pathwaysbc.ca/specialists/1134 who requested to be linked to his profile so he could keep it up to date, but I could not do this for him.  Therefore he has never activated his Access key.
  
Are these changes technically possible?  And how difficult and time-consuming would they be?  

===============

Also related: #212 (categorization bug) and #235 (search)

**Comment by bgracie** [comment]
background for this issue from Russell:

We ultimately decided that, if a specialist took referrals directly, they were considered to have “responded to survey”, marked as “accepting new patients” (or whatever relevant status describes them), and then they would be given an office in the clinic, thereby picking up the clinics address (and thus city).

If they, however, take referrals through a pooled process at a clinic, where a GP simply referred to the clinic, and not the specific specialist, we marked them as “only works out of hospitals or clinics” (and then we put these hospitals / clinics as associations, and direct users to refer to them through said hospitals or clinics. We then also use these hospitals / clinics as the source of their city, as this is where they work.

So, going back to the first case: if a specialist is “responded to survey” and has some offices listed, they may also have some clinic or hospital associations that they don’t have offices listed in. Right now these are then assumed to be the places that they don’t see patients in, and thus we aren’t listing them as cities they work in. That said, perhaps this is incorrect, as we might have a specialist who sees patients in a few offices but also takes referrals directly throughs some clinics – a hybrid of “responded to survey” and “only works out of hospitals and clinics” – where these clinic cities should show up. Basically, we don’t have a perfect way to capture all the edge cases.

In summary: I’m not sure it’s a perfect decision either way, and is a bit of a complex issue. It’s inconsistent for administrators now, but errs on the side of a conservative list of cities we know that specialists take patients in. Making it consistent would mean we might add a few cities for people who only have privileges somewhere, but don’t actually take / see patients there.

I think it would be interesting to generate a list of specialists who are “responded to survey” and who would have cities added if you included their clinic privileges and hospitals attendances. It may be that this is a very small number (as presumably most work in hospitals and clinics in the same general area as their practice), and then the decision would be easier.

**Comment by bgracie** [comment]
IMO there's at least three things going on in "categorization": 
- whether or not they responded to the survey
- where they work
- where they take referrals

If we don't break these things up to be modifiable separately we're going to face ongoing pain.

**Comment by bgracie** [comment]
- "where they work", we have from the rest of the form (offices, privileges, attendances)
- "where they take referrals" could be its own dropdown, along with "where/whether they were surveyed or not"

**Comment by bgracie** [comment]
Given that we would end up with more possible combinations of values for these options than we have now (e.g. "responded to survey" but also only works out of their own office), we would have to think through how these new combinations affect the things that previously depended on the old complicated "categorization" (e.g. location and referral forms).

**Comment by bgracie** [comment]
A solution along these lines should also be able to address the problem of wanting to allow entry of specialists (with their proper location, instead of "no division") without showing them to users.

**Comment by KelseyDH** [comment]
Follow up on this matter.  Last Saturday Brian, Ron, Russell, Tracy and I met to discuss how to handle Specialists who hold the categorization of "only work out of hospitals and clinics" (with a blue arrow).

We looked over Tracy L's proposal to update this categorization and examined the problem using the problem example given to us by Vancouver division of handling data entry for Dr. REDACTED.  The discussion was productive and I agreed to summarize a few points in relation to this matter.

Decisions from the group's meeting:


**1.   Specialist profile hiding too much information:**

Russell agreed the current behaviour of the system removing information about a Specialist after they are categorized as "only works out of hospitals and clinics" may be excessive -- though he did note that the "Additional Details" box can still be used for adding more information if a specialist is categorized as "only works out of hospitals and clinics".  

As such, Russell has agreed to mock up a better way of displaying information about Specialists when their categorization is changed to "only works out of hospitals and clinics" so that the stripping down of the Specialist's profile information is not so drastic.  After these changes he believes Dr. REDACTED would be able to have the categorization "only works out of hospitals and clinics" without the change looking so visually jarring.

**2.   Specialist edit box removing too many edit form sections:**

The group reviewed the edit form sections being hidden when "only works out of hospitals and clinics" is selected as a categorization, and believe most of these form fields should continue to remain hidden, since they pertain to referral information that should be entered for that Specialist at the office or Clinic level.  

It was also noted that for difficult edge cases another categorization does exist called *"accepting referrals in an existing office"* (currently only available in Family Practice) that could be applied to other Specialties if there is agreement.   One idea was for the wording to be updated to say:  *"But they also have an office below where they do not accept referrals but they may see patients there for ongoing care"*

Furthermore, Russell did not believe it was a good idea to remove the last Location option "In a new office in an existing Clinic" as he believed this entry still had use.   He said you would use this option when a specialist has a brand new office and is in a larger clinic.

Finally, we agree that Language should be added back.


**3.   Referral Form no longer being available:**

We agree that this is a problem, but we propose that instead of tacking on Referral Forms to these types of specialists (and creating further duplication of referral form uploads), that we should instead modify the referral form system to be more robust by allowing all forms to be independently linked or delinked from any specialist or clinic.  This way Referral Forms can be uploaded and maintained separate from the entity they are uploaded for.  However, this will require a little bit of coding work to implement so we recommend this comes up as an independent change request next meeting for prioritization.

**4.   Users being unable to control the profile**

The group agreed this should be fixed.  However technically user editing permissions within the system are defined at the specialist office level, rather than at the specialist profile level, and this explains why this hasn't been implement.  It was agreed that the system should allow a user to control/edit a specialist if they are categorized with as "only works out of hospitals and clinics", however it was also acknowledged that it would take a moderate amount of programming work to implement.  So it was recommended that this change request be sent to the next user group meeting for consideration.

**Comment by KelseyDH** [comment]
Investigation finished, progress will now be tracked through the follow up issues above.
